File: 2021/01/task-02/main.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    char str[8];
    int increases = 0;
    unsigned long int depths[] = {0, 0, 0};
    int j;

    for (j = 0; fgets(str, sizeof str, stdin) != NULL; j++) {
        unsigned long int depth = strtoul(str, NULL, 10);

        if (j >= 3 && depth > depths[0]) {
            increases++;
        }

        depths[0] = depths[1];
        depths[1] = depths[2];
        depths[2] = depth;
    }

    printf("%d\n", increases);

    return 0;
}

File: 2021/01/task-02/src/main.rs
use std::io::{stdin, BufRead};

fn main() {
    let mut window: Vec<u16> = vec![];
    let mut increases = 0u16;

    stdin()
        .lock()
        .lines()
        .filter_map(|line_result| line_result.ok())
        .filter_map(|line| line.parse::<u16>().ok())
        .for_each(|depth| {
            window.push(depth);

            if window.len() > 3 && depth > window.remove(0) {
                increases += 1;
            }
        });

    println!("{}", increases);
}
